File: Binary search /Square root of a number(using BS).java
//https://www.geeksforgeeks.org/problems/square-root/0

// Function to find square root
// x: element to find square root
class Solution {
    long floorSqrt(long n) {
        // Your code here
    long low =0 , high =n;
    while(low <= high){
         long mid = (low + high)/2;
         long val = mid* mid;
        if(val <= n){
            low = mid+1;
        }
        else{
            high = mid-1;
        }
    }
    return high;
    }
}

File: Binary search /Square Root of a number(using BS).cpp
//https://www.naukri.com/code360/problems/square-root-integral_893351
int floorSqrt(int n)
{
    // Write your code here.
    int low =0 , high =n;
    while(low <= high){
        long long mid = (low + high)/2;
        long long val = mid* mid;
        if(val <= n){
            low = mid+1;
        }
        else{
            high = mid-1;
        }
    }
    return high;
}
